src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupService.java

```diff
@@ -612,4 +622,222 @@ private Optional<User> findStudent(StudentDTO studentDto, String studentCourseGr
+            return getDayOfWeekString(tutorialGroup.getTutorialGroupSchedule().getDayOfWeek());
+        }
+        return getDayOfWeekString(8); // Default to "None"
+    }
+
+    /**
+     * Returns the requested field (start time, end time, location) from the tutorial group's schedule.
+     * If the schedule or the requested field is null, returns an empty string.
+     */
+    private String getScheduleField(TutorialGroup tutorialGroup, ScheduleField field) {
+        if (tutorialGroup.getTutorialGroupSchedule() != null) {
+            switch (field) {
+                case START_TIME -> {
+                    return getValueOrDefault(tutorialGroup.getTutorialGroupSchedule().getStartTime());
+                }
+                case END_TIME -> {
+                    return getValueOrDefault(tutorialGroup.getTutorialGroupSchedule().getEndTime());
+                }
+                case LOCATION -> {
+                    return getValueOrDefault(tutorialGroup.getTutorialGroupSchedule().getLocation());
+                }
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Returns the requested field (registration number, first name, last name) from the student.
+     * If the student or the requested field is null, returns an empty string.
+     */
+    private String getStudentField(User student, StudentField field) {
+        if (student != null) {
+            switch (field) {
+                case REGISTRATION_NUMBER -> {
+                    return getValueOrDefault(student.getRegistrationNumber());
+                }
+                case FIRST_NAME -> {
+                    return getValueOrDefault(student.getFirstName());
+                }
+                case LAST_NAME -> {
+                    return getValueOrDefault(student.getLastName());
+                }
+            }
+        }
+        return "";
+    }
+
+    // Enum to represent the different fields of the schedule
+    private enum ScheduleField {
+        START_TIME, END_TIME, LOCATION
+    }
+
+    // Enum to represent the different fields of the student
+    private enum StudentField {
+        REGISTRATION_NUMBER, FIRST_NAME, LAST_NAME
+    }
+
+    /**
+     * Converts a numeric representation of a day of the week to its corresponding String name.
+     * This method is used primarily for CSV export purposes.
+     *
+     * @param dayOfWeek the numeric representation of the day of the week, where Monday = 1 and Sunday = 7.
+     * @return a String representing the name of the corresponding day of the week.
+     * @throws IllegalArgumentException if the provided dayOfWeek is not within the range 1 to 7.
+     */
+    public String getDayOfWeekString(int dayOfWeek) {
+        if (dayOfWeek < 1 || dayOfWeek > 8) {
+            throw new IllegalArgumentException("Invalid day of the week: " + dayOfWeek);
+        }
+        String[] days = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday", "None" };
```

Comment:
Why does `8` map to `"None"`? Maybe you could add a comment explaining

Reply:
done 👍